### DISCUSSION TITLE
Fine-tuning to new task with 🤗

### DISCUSSION THREAD

**Discussion by npbeck** [google-research/text-to-text-transfer-transformer#766] [Q&A] [answered]
I would like to fine-tune T5 to a new task with control codes, in a supervised manner.

The input would look like this:
`"<new_task_name> <variable_control_code>: <input_text>"`

Also, I would like to use the [🤗 wrapper](https://huggingface.co/transformers/model_doc/t5.html#t5forconditionalgeneration) for this. 
I am wondering if I need to implement a new `task` outside of the 🤗 wrapper and inform the model explicitly of this task. Or can I just feed my training data to T5 in the format above? Does it make a difference?

Thanks in advance!

**Comment by craffel** [comment, marked as answer]
The advantage of using a t5.data.Task is that it makes it easy to handle things like encoding/decoding with the vocabulary, evluation, etc. But yes, you can also just feed the text in directly and use HF's tools to tokenize/detokenize/evaluate/etc.
